packages/fewer/src/Migration/index.ts

```diff
@@ -0,0 +1,28 @@
+export interface Migration {
+  up: string[];
+  down?: string[];
+}
+
+interface ChangeMigrationDefinition<T> {
+  change: (m: Pick<T, Exclude<keyof T, 'rawsql'>>) => void;
+}
+
+interface UpDownMigrationDefinition<T> {
+  up: (m: T) => void;
+  down: (m: T) => void;
+}
+
+interface IrreversibleMigrationDefinition<T> {
+  up: (m: T) => void;
+  irreversible: true;
+}
+
+export type MigrationDefinition<T> = ChangeMigrationDefinition<T> | UpDownMigrationDefinition<T> | IrreversibleMigrationDefinition<T>;
+
+export function isChangeMigration<T>(definition: MigrationDefinition<T>): definition is ChangeMigrationDefinition<T> {
```

Comment:
You could probably avoid these functions if you put a static `type` field on all of the migrations and set it to some enum value.